Lib/test/test_builtin.py

```diff
@@ -1511,6 +1513,126 @@ def test_construct_singletons(self):
             self.assertRaises(TypeError, tp, 1, 2)
             self.assertRaises(TypeError, tp, a=1, b=2)
 
+
+class TestBreakpoint(unittest.TestCase):
+    def test_breakpoint(self):
+        with patch('pdb.set_trace') as mock:
+            breakpoint()
+        mock.assert_called_once()
+
+    def test_breakpoint_with_breakpointhook_set(self):
+        call_status = 'Not called'
+        def my_breakpointhook():
+            nonlocal call_status
+            call_status = 'Called'
+        try:
+            sys.breakpointhook = my_breakpointhook
+            breakpoint()
+        finally:
+            sys.breakpointhook = sys.__breakpointhook__
+        self.assertEqual(call_status, 'Called')
+
+    def test_breakpoint_with_breakpointhook_reset(self):
+        call_status = 'Not called'
+        def my_breakpointhook():
+            nonlocal call_status
+            call_status = 'Called'
+        try:
+            sys.breakpointhook = my_breakpointhook
+            breakpoint()
+        finally:
+            sys.breakpointhook = sys.__breakpointhook__
+        self.assertEqual(call_status, 'Called')
+        with patch('pdb.set_trace') as mock:
+            breakpoint()
+        mock.assert_called_once()
```

Comment:
Would it be possible to restore sys.breakpointhook to its original value at the end of the test?

Reply:
Yep, see above.